### DISCUSSION TITLE
Strange behavior - First LED lights white - Wemos D1 D4Pin always high level

### DISCUSSION THREAD

**Discussion by TPPTim** [awawa-dev/HyperHDR#165] [Q&A]
Hello together & developer,
first i wanna say thank you for sharing this grate open source Software with the World.
I try to install it in my living room behind my TV, but I have a strange problem.

If I connect the Wemos D1 to a USB port, the first LED in the **SK6812** row lights up white. The rest is dark.
A measurement with an Oscilloscope shows me that the D4 Pin, where I connected the D-in of my LED strip) is allway at high 3,3V Level. This is my "LevelShifter" LED at 4,3V.
If i start HyperHDR right now, nothing will change.
HyperHDR is running on my Windows10 machine at **Version 17.0.0**.
It grabs the Windows Screen and send the info @2Mb Serial to a **Wemos D1 (CH340C+ESP8266)**. I used this file: **HyperSerialEsp8266.ino.d1_mini_SK6812_NEUTRAL_WHITE_FIRSTLED_DISABLED.bin**
I can measure the Serial Data at the Rx Pin of the ESP module. Seems correct so far.

Can you help me out? I don't know what's wrong with it. To test the Wemos D1 module, I flashed the popular WLED.ino project and tested the ESP and the LEDs via my smartphone. It worked well, so the Hardware is ok.

I flashed the Software via **ESP Home** flasher without problems. 

Thanks for reading my text and thinking about it :)

Regards Tim

**Comment by awawa-dev** [comment]
Hi Tim
As you can see in latest HyperSerialEsp8266 sources, it doesn't write any data to LED strip when no data arrives. Only when full frame from HyperHDR arrives to HyperSerialEsp8266 and you are using FIRSTLED_DISABLED version, then it writes black color for first LED. But 
the data must arrive and recently there are various problems with counterfeit ch340 . So to be sure you must run HyperHDR for a while and then shut it down and check out statistics with ```screen```  or other serial port client connected at 2Mb to wemos (it should tell if the HyperSerialEsp8266 received something during that session, do not restart it during that procedure).

Constant 1 on D4 should not cause such reaction from that single level shifter LED.
![obraz](https://user-images.githubusercontent.com/69086569/144928004-48fe68cd-8f25-472a-9dbc-47ce6a8e7923.png)
sk6812 needs zero level for the protocol to be recognized and synchronized.
Regards
Awawa

**Comment by TPPTim** [comment]
Hey Awawa,

thanks for your fast response.
I tried it as you suggested and this is my output:
![grafik](https://user-images.githubusercontent.com/95652405/144932244-475ee7d1-3b88-478e-8da4-41c28335a733.png)

I tried different baud rates, slowly decreasing:
![grafik](https://user-images.githubusercontent.com/95652405/144932395-39cf9b60-19b1-48fb-9c9d-ce22b795f65f.png)


I restarted the Wemos D1, but still get the same:
![grafik](https://user-images.githubusercontent.com/95652405/144932512-abe86045-18c2-4ac4-a81b-35deed41f28c.png)


Does the ESP send me basic stuff after a clean flashing process without putting afterwards data trough HyperHDR, to check if the serial connection works fine?
Or your right and the C340C is not original.... these chines copies...

Regards
Tim

**Comment by awawa-dev** [comment]
You should see some 'normal messages from HyperSerialEsp8266 (version, color calibration etc)  when restarting it with a button , even without HyperHDR working in meantime. I'm afraid it could be the case:
https://github.com/awawa-dev/HyperHDR/issues/129#issuecomment-940413908
https://github.com/awawa-dev/HyperHDR/discussions/97#discussioncomment-1213991
Linux dmesg could tell the true. But unfortunately the communication is broken anyway.
